src/USER-CONP/ewald_conp.cpp

```diff
@@ -1810,32 +1871,36 @@ void EwaldConp::compute_matrix(bigint *imat, double **matrix) {
 
         // anyway, use local sn and cs for simplicity
 
-        kx = kxvecs[k];
-        ky = kyvecs[k];
-        kz = kzvecs[k];
-        sign_ky = (ky > 0) - (ky < 0);
-        sign_kz = (kz > 0) - (kz < 0);
+        int const kx = kxvecs[k];
+        int const ky = kyvecs[k];
+        int const kz = kzvecs[k];
+        int const sign_ky = (ky > 0) - (ky < 0);
+        int const sign_kz = (kz > 0) - (kz < 0);
 
-        cos_kxky = cs[kx][0][i] * cs[ky][1][i] - sn[kx][0][i] * sn[ky][1][i];
-        sin_kxky = sn[kx][0][i] * cs[ky][1][i] + cs[kx][0][i] * sn[ky][1][i];
+        double cos_kxky =
+            cs[kx][0][i] * cs[ky][1][i] - sn[kx][0][i] * sn[ky][1][i];
+        double sin_kxky =
+            sn[kx][0][i] * cs[ky][1][i] + cs[kx][0][i] * sn[ky][1][i];
 
-        cos_kxkykz_i = cos_kxky * cs[kz][2][i] - sin_kxky * sn[kz][2][i];
-        sin_kxkykz_i = sin_kxky * cs[kz][2][i] + cos_kxky * sn[kz][2][i];
+        double const cos_kxkykz_i =
+            cos_kxky * cs[kz][2][i] - sin_kxky * sn[kz][2][i];
+        double const sin_kxkykz_i =
+            sin_kxky * cs[kz][2][i] + cos_kxky * sn[kz][2][i];
 
         // global indexing  csx_all[kx+j*(kxmax+1)]  <>  csx_all[kx][j]
 
-        kxj = kx + j * (kxmax + 1);
-        kyj = abs(ky) + j * (kymax + 1);
-        kzj = abs(kz) + j * (kzmax + 1);
+        int const kxj = kx + j * (kxmax + 1);
+        int const kyj = abs(ky) + j * (kymax + 1);
+        int const kzj = abs(kz) + j * (kzmax + 1);
 
         cos_kxky =
             csx_all[kxj] * csy_all[kyj] - snx_all[kxj] * sny_all[kyj] * sign_ky;
         sin_kxky =
             snx_all[kxj] * csy_all[kyj] + csx_all[kxj] * sny_all[kyj] * sign_ky;
 
-        cos_kxkykz_j =
+        double const cos_kxkykz_j =
```

Comment:
Ist es OK in einem Loop jedesmal einen double const zu initialisieren?

Reply:
Es funktioniert jedenfalls. Ob das klug ist weiß ich nicht.

Reply:
Mh, OK, check nochmal, ob es so Sinn macht, oder ob man lieber einmal ausserhalb des Loops cos_kxkykz etc definiert und sie nachher nur überschreibt... Könnte u.U. schneller sein.